src/lib.rs

```diff
@@ -1,21 +1,96 @@
 //! The project that will henceforth be known as `starling`.
-
+//!
+// `error_chain!` can recurse deeply
+#![recursion_limit = "1024"]
 #![deny(missing_docs)]
 #![deny(missing_debug_implementations)]
+#![deny(unsafe_code)]
+// Annoying warning emitted from the `error_chain!` macro.
+#![allow(unused_doc_comment)]
 
+extern crate clap;
+#[macro_use]
+extern crate error_chain;
 extern crate futures;
 extern crate tokio_core;
 
+pub mod cli;
```

Comment:
Let's start this off with proper separation and not include this in `lib.rs`: the library shouldn't know about arg parsing. (Yes, I realize this is all scaffolding, but this is strictly code we'd remove from this file later on anyway.)